### DISCUSSION TITLE
Quarkus always using the same thread for all requests on the same route?

### DISCUSSION THREAD

**Discussion by newk5** [quarkusio/quarkus#29231] [Q&A] [answered]
So I'm using Resteasy Reactive and I was under the impression that quarkus would spawn a new thread when the current thread is blocked and this seems to be true when different routes are hit, but when hitting the same route, it keeps using the same thread.

For example I have 2 routes:
 
- GET /hello (slow operation taking 10 seconds)
- GET /hello/hello2 (quick operation returning string)

If I hit **/hello** and then imediately hit **/hello/hello2** I get an instant response from the 2nd route while the first route is still processing, which is great!

However if I hit the first route twice ( **/hello**  ) the second send request will not even start being processed until the first one finishes. This means that all it takes, is 1 slow request to  **/hello**  for this route to be blocked for all other clients. 

Any way to fix this?

Here's a small reproducer:

```java
@Path("/hello")
public class GreetingResource {

    static AtomicLong i = new AtomicLong(0);

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() throws InterruptedException {
        Long c = i.incrementAndGet();

        if (c == 1) {
            System.out.println("SLOW OPERATION IN PROGRESS");
            Thread.sleep(10000);

        } else {
            System.out.println("QUICK OPERATION");
        }
        return Thread.currentThread().getName();
    }
}
```

Just hit the /hello route twice, the second request should return instantly, but you will see the second request will be blocked by the first

(Using quarkus 2.14.0.Final)

**Comment by geoand** [comment, marked as answer]
What you are describing is not the intended behavior and it is also **not** what I am seeing.
When I tried your example, everything worked as expected.

**Reply by newk5** [reply]
Did you try the exact code I posted? I can reproduce it pretty consistently, though I admit I only did the test in dev mode.

**Reply by geoand** [reply]
Yes, the exact same piece of code you posted.

**Reply by newk5** [reply]
Ok then I guess I'll try create a small reproducer project and post here

**Reply by newk5** [reply]
@geoand did you test on firefox or chrome? This is very strange but I am now noticing this issue only seems to happen on chrome. When I test on firefox everything works as expected. Also what OS are you on? I'm on W11

**Reply by geoand** [reply]
It seems like this is a Chrome issue, basically Chrome seems to not send the next request before the previous one has completed.
You can clearly see this behavior when using capturing the traffic with Wireshark.

Firefox does not exhibit this behavior.

**Reply by newk5** [reply]
Hmm yea, I think you're right. I tested on edge and seems like it's also happening there. I guess all chromium based browsers are affected. 

Some videos I made:

Firefox - 

https://user-images.githubusercontent.com/32560510/202148771-e47c22cd-1d5f-4ef8-a326-b5e465de4d1e.mp4

Chrome - 

https://user-images.githubusercontent.com/32560510/202148945-f61ee910-3d2e-4073-8078-211bf88d8fb6.mp4

**Reply by newk5** [reply]
After further testing I am now noticing that if I make the 2nd request on a completely new chrome window, the issue does not happen. It happens only when the 2nd request is made on a different tab, still a bug in my opinion, but not quarkus related anyway

**Reply by geoand** [reply]
I have not looked into it, but I am guessing that there is some good reason why Chrome does this although I can't guess why
